File: inventory/src/main/java/com/inventory/entity/PurchaseOrder.java
package com.inventory.entity;

import jakarta.persistence.*;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.AllArgsConstructor;
import java.time.LocalDateTime;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "purchase_orders")
public class PurchaseOrder {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "item_id", nullable = false)
    private Item item;

    @Column(nullable = false)
    private Integer quantity;

    @Column(nullable = false)
    private LocalDateTime orderDate;

    @Column(nullable = true)
    private LocalDateTime arrivalDate;

    @Column(nullable = true)
    private String trackingNumber;

    @Column(nullable = false)
    private Boolean arrived = false;

    @Column(nullable = true)
    private String createdBy;

    @Column(nullable = true)
    private String arrivedBy;

    @Column(nullable = false)
    private LocalDateTime createdAt;

    @Column(nullable = false)
    private LocalDateTime updatedAt;

    @PrePersist
    protected void onCreate() {
        createdAt = LocalDateTime.now();
        updatedAt = LocalDateTime.now();
        if (orderDate == null) {
            orderDate = LocalDateTime.now();
        }
    }

    @PreUpdate
    protected void onUpdate() {
        updatedAt = LocalDateTime.now();
    }

    public void markAsArrived(String arrivedBy) {
        this.arrived = true;
        this.arrivalDate = LocalDateTime.now();
        this.arrivedBy = arrivedBy;
    }
} 

File: inventory_frontend/app/types/purchaseOrder.ts
export interface PurchaseOrder {
  id: number;
  itemId: number;
  itemName: string;
  quantity: number;
  orderDate: string;
  arrivalDate?: string;
  trackingNumber?: string;
  arrived: boolean;
  createdBy?: string;
  arrivedBy?: string;
  createdAt: string;
  updatedAt: string;
}

export interface PurchaseOrderRequest {
  itemId: number;
  quantity: number;
  orderDate?: string;
  trackingNumber?: string;
}

export interface PurchaseOrderCreateRequest {
  quantity: number;
  trackingNumber?: string;
} 